web3/providers/async_base.py

```diff
@@ -36,6 +36,7 @@ class AsyncBaseProvider:
     # a tuple of (all_middlewares, request_func)
     _request_func_cache: Tuple[Tuple[Middleware, ...], Callable[..., RPCResponse]] = (None, None)
 
+    ccip_read_calls_enabled: bool = True
```

Comment:
semantics nit to take or leave: I know we talked about this some on Discord, but seeing `ccip_read_calls_enabled` next to `ccip_read_enabled` doesn't give me a good idea of the difference between the two. Maybe something like `global_ccip_read_enabled` would be more descriptive? /cc @marcgarreau :)

Reply:
No expectation as of now. We provide a pretty easy way to handle [preflighting](https://web3py.readthedocs.io/en/latest/examples.html#ccip-read-support-for-offchain-lookup) if need be at the moment.

Reply:
down the future-proofing semantics rabbit hole:

i think it comes down to: if we were to add ccip read support beyond call, would you prefer
a) more granularity, e.g., more flags - `ccip_read_calls_enabled`, `ccip_read_tx_enabled`, etc. or
b) everything covered by a single flag, e.g., `ccip_read_enabled` or `global_ccip_read_enabled`

with limited context, my instinct is `b`.

Reply:
If it is expected to be the master on/off switch for CCIP Read functionality, then I would agree adding `global` helps clarify, and the inclusion of `call` in the name is less important.

Reply:
For sure, I'd prefer that too of we went that route. We talked about not handling transactions automatically since it gets webbed up by our middleware triggering `eth_estimateGas` if there is no `gas` field on the original transaction, etc... and this mainly a read functionality atm.

I think the main goal here is to distinguish the provider flag from the `eth_call` flag since we don't have a flag that is all-encompassing.

Reply:
I think global works as long as the documentation clarifies we only turn this on for calls and that seems to be the consensus 👍🏼.